src/main/java/com/xceptance/xlt/api/htmlunit/LightWeightPage.java

```diff
@@ -128,37 +128,71 @@ private Charset determineContentCharset()
     {
         if (response != null)
         {
+            /*
+             * TODO: I would love to replace all this code with a simple "response.getContentCharset()" as it is much
+             * more elaborate and robust. Unfortunately, that method behaves (slightly) different. Maybe we can do this
+             * with the next major version.
+             */
+
             // 1st: get value of content-type response header
-            String charsetName = StringUtils.substringAfter(response.getResponseHeaderValue("content-type"), "charset=");
-            if (StringUtils.isEmpty(charsetName))
+            String charsetName = getCharsetNameFromContentTypeHeader(response);
+            if (StringUtils.isBlank(charsetName))
             {
                 final String content = response.getContentAsString(StandardCharsets.ISO_8859_1);
-                if (!StringUtils.isEmpty(content))
+                if (StringUtils.isNotBlank(content))
                 {
-                    // 2nd: get the encoding attribute from a potential <?xml?>
-                    // header (in case of XHTML)
+                    // 2nd: get the encoding attribute from a potential <?xml?> header (in case of XHTML)
                     charsetName = RegExUtils.getFirstMatch(content, "<\\?xml.*? encoding=\"(.+?)\".*?\\?>", 1);
```

Comment:
Just wondering if there is any relevant difference to use `encoding=\"([^\"]*)\"` instead of `encoding=\"(.*?)\"` in regex.